File: 5/python/main.py
from string import punctuation

def main() -> None:
    word = 'Слава Україні!'.lower()
    numbers = '004014063076165102 022032003012005!'

    alphabet = 'абвгґдеєжзиіїйклмнопрстуфхцчшщьюя'

    index = 1
    for symbol in word:
        if symbol in punctuation or symbol.isspace():
            print(symbol, end='')
            index = 1
        else:
            print('{:03}'.format((alphabet.find(symbol) + 1)*index), end='')
            index+=1
    print()

    index = 0
    step = 1
    while(index < len(numbers)):
        if numbers[index] in punctuation or numbers[index].isspace():
            print(numbers[index], end = '')
            step = 1
            index += 1
        else:
            print(alphabet[(int(numbers[index: index + 3]) - 1) // step], end ='')
            step += 1
            index += 3
    print()

if __name__ == "__main__":
    main()
        

File: 5/js/main.js

const main = () =>{
    let word = 'Слава Україні!'.toLowerCase();
    let numbers = '004014063076165102 022032003012005!';
    let alphabet ='абвгґдеєжзиіїйклмнопрстуфхцчшщьюя';

    let index = 1;
    for(symbol of word){
        if(symbol.charCodeAt(0) <= 47){
            index = 1;
            process.stdout.write(symbol);
        }
        else{
            process.stdout.write( ((alphabet.indexOf(symbol) + 1)*index).toString().padStart(3, '0') );
            ++index;
        };
    };

    console.log();
    
    index = 1;
    for(let i = 0; i < numbers.length; ){
        if(numbers[i].charCodeAt(0) <= 47){
            process.stdout.write(numbers[i]);
            index = 1;
            ++i;
        }
        else{
            process.stdout.write(alphabet[(Number(numbers.slice(i, i+3) - 1 ) / index|0)]);
            i+=3;
            ++index;
        };
    }
    console.log();
};

main();